### PR TITLE
Add TypeScript testing intro project

### PR DESCRIPTION
**Pull request by brnkrygs**
*Issue #4*

*Description of changes:*
Add TypeScript testing intro project, based on basic `sam init` TypeScript project. Includes unit and integration test samples.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by roman-boiko — APPROVED** [review]
Overall LGTM

File: typescript-test-samples/typescript-test-intro/list-buckets/tests/unit/list-buckets.test.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        });
+    });
+});
+
+/**
+ * Builds an empty API Gateway Proxy Event input object, with default values
+ * @returns {APIGatewayProxyEvent}
+ */
+function getEmptyAPIGatewayProxyEvent(): APIGatewayProxyEvent {
+    const result: APIGatewayProxyEvent = {
```
Comment (dancfox) [inline typescript-test-samples/typescript-test-intro/list-buckets/tests/unit/list-buckets.test.ts:114]: In the Python project I had initially created a proxy event in a function. However later we refactored to put this event into the events folder as a JSON object. (see this: https://github.com/aws-samples/serverless-test-samples/blob/fad4659150c06ca0acc4b63d28c3be0a1a30729a/python-test-samples/apigw-lambda/tests/unit/mock_test.py#L14). I think the JSON approach is cleaner. What do you think?

**Review by dancfox — CHANGES_REQUESTED** [review]

File: typescript-test-samples/typescript-test-intro/list-buckets/tests/unit/list-buckets.test.ts (hunk 1 above)
Comment (brnkrygs) [reply typescript-test-samples/typescript-test-intro/list-buckets/tests/unit/list-buckets.test.ts:114]: Did it this way to keep the TypeScript type engine/compiler happy. I'll see if we can do a clean read+cast off of a JSON file to get the best of both worlds.

Comment (brnkrygs) [reply typescript-test-samples/typescript-test-intro/list-buckets/tests/unit/list-buckets.test.ts:114]: Also concerned about node imports pulling the JSON struct in as a singleton, then having the unit tests clobber each other. The way we have it now, its a Factory, so each test gets its own event instance. Will investigate this more deeply too.

**Comment by brnkrygs** [comment]
Re-opening after sync and addressing @dancfox comments.

**Review by dancfox — APPROVED** [review]

**Pull request merged**